### PR TITLE
Create Quality Assurance table for Delete Task API #757

### LINKED ISSUE #757: Create Quality Assurance table for Delete Task API
## Description
As a developer,
I need to know what are the test cases for a certain API,
so that I can write new unit tests.

As a tester,
I need to know what are the test cases for a certain API,
so that I can perform manual tests on the API.

The task is to come up and add a quality assurance table for "Delete Tasks" API to the docs in the /docs folder.

Additionally, add all tasks' quality assurance tables under one heading. Currently, the Update task is under heading "Mentorship Relation" and Create Tasks under heading "Tasks". I suggest make the subheading Tasks under Mentorship Relation and add all Tasks' API under it.

#### Note: This issue is labeled for Open Source Hack.
I can assign this issue to you on September 1st as the official opening date for Open Source Hack. Please free to check out other issues in the meantime. Program website: https://anitab-org.github.io/events/open-source-hack/

## Mocks
As done under [docs/quality-assurance-test-cases.md](https://github.com/anitab-org/mentorship-backend/blob/develop/docs/quality-assurance-test-cases.md)

mentorship-backend

### Update [Required]
- [ ] Add test cases for Delete Task API.
- [ ] Add all tasks' quality assurance tables under one heading.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentors.

## Estimation
[4+] hours

### PR DESCRIPTION
**Pull request by dowenrei**
### Description
Added test cases for Delete Task API and create a heading "Task" for Create, Update and Delete Task.
Fixes #757 

### Type of Change:
- Quality Assurance

**Code/Quality Assurance Only**
- [/ ] My changes generate no new warnings

### CODE REVIEW

File: docs/quality-assurance-test-cases.md [hunk 1]
```diff
@@ -173,7 +185,16 @@ This document contains some examples of test cases for each feature implemented
 | Logged in user tries to complete a task from an non existing request (The request  exists in a different relationship) | Fail |
 | Logged in user tries to complete a task from a request which is not in the ACCEPTED state (as a mentor or as a mentee) | Fail |
 
+### Delete Task
 
+**Service:** DELETE /mentorship_relation/{request_id}/task/{task_id}
+| Test Case       | Outcome |
```
Comment (paritoshsinghrahar) [inline docs/quality-assurance-test-cases.md]:
```
| ------------- | ------------- |
| Logged in user delete a task from an existing request in the ACCEPTED state| Success |
| Logged in user tries to delete a task whose id does not exist from an existing request in the ACCEPTED state (the task does not exist) | Fail |
| Logged in user tries to delete a task from a non-existing request (Mentorship relation does not exist) in the ACCEPTED state | Fail |
| Logged in user tries to delete a task from a request which the user is not a mentor nor a mentee | Fail |
| Non-Logged in user tries to delete a task from an existing request in the ACCEPTED state  | Fail |
```
A nitpick. Can you uniformly space out the table in markdown for uniformity with other tables?

Comment (dowenrei) [reply docs/quality-assurance-test-cases.md]: When I check on github, it looks fine. Do you mean changing the spacing at the code (as shown below)?
DeleteTask

Comment (paritoshsinghrahar) [reply docs/quality-assurance-test-cases.md]: Yes, precisely.

**Review by paritoshsinghrahar — APPROVED** [review]

**Comment by devkapilbansal** [comment]
@dowenrei can you please resolve the merge conflicts??

**Comment by devkapilbansal** [comment]
@dowenrei are you working on this?

**Comment by dowenrei** [comment]
> @dowenrei can you please resolve the merge conflicts??

I'm not authorized to solve this conflict.

**Comment by devkapilbansal** [comment]
> > @dowenrei can you please resolve the merge conflicts??
> 
> I'm not authorized to solve this conflict.

You can since this is a pull request from your fork. Just make the necessary changes locally and push them

**Comment by vj-codes** [comment]
@dowenrei hey any updates?

**Comment by vj-codes** [comment]
@dowenrei closing this due to inactivity
Let us know if you would like to continue again and thank you for your contribution :)

**Pull request closed without merging**